js/src/logger.ts

```diff
@@ -1147,6 +1147,13 @@ export function startSpan(args?: StartSpanOptionalNameArgs): Span {
   const name =
     (nameOpt ?? iso.getCallerLocation()?.caller_functionname) || "root";
   const parentSpan = currentSpan();
+
+  if (!parentSpan) {
+    throw new Error(
+      "Cannot call startSpan() from outside a trace. Please wrap this code in a traced() callback."
```

Comment:
I think it's always a bug if `parentSpan` is undefined. The ISO should be initializing it to a noop span.

Reply:
discussed offline -- we need to dig into this further